### DISCUSSION TITLE
Enhancements

### DISCUSSION THREAD

**Discussion by Wolf242** [nealmick/bb#41] [Ideas]
Here are some strategies:

Expand Player Performance Data: Your model currently focuses on the top seven players based on playing time. Consider including data from the last 5 and 10 games for these players. This will add a more immediate and current performance aspect to your predictions.

Incorporate Points Per Minute and Quarter Analysis: Your model already includes detailed statistics like field goals, assists, blocks, etc. Enhance this by calculating points per minute for players and conducting a quarter-by-quarter analysis. This will add depth to your player and game predictions.

Refine Team Projection Metrics: You already use advanced metrics, but ensure to include pace, offensive and defensive efficiency in your calculations. This will provide a more nuanced view of each team's capabilities.

Enhanced Evaluation Metrics: Your model already evaluates win/loss percentage, spread win/loss percentage, and margin-based evaluations. Consider refining these metrics to account for the additional data points and simulation outcomes you'll now be incorporating.

Iterative Testing and Validation: With these new integrations, it's crucial to continuously test and validate the model. This can include back-testing with historical data and adjusting the model based on real-world performance.

By carefully integrating these aspects, your model can become more dynamic, taking into account both the current form of players and teams as well as their longer-term performance trends. This holistic approach can significantly enhance the predictive power and accuracy of your sports betting strategy.

**Comment by nealmick** [comment]
these are really good ideas, and probably buildable too.  ima reply like fully to this tomorrow because i dont have the energy right this second.  But good ideas!

**Comment by nealmick** [comment]
okay i reply now with this, but may still write more tomorrow.... 
Planning on adding more game history not sure where the limits to that are... the main difficulty here is the api requests.... we have to get all that data when someone clicks and provide a relatively quick page load.  first loads we have to request all the data that isnt already cached.... and its local cache so its not like we share between installs and the live server.  Each machine has its own cache that it can build up.... if i focus just on the live server, it become much easier to guaranteed fast loads because everyone shares the cache... and its more likely that someone has already requested some portion of the data needed.... where as locally its just you.  The main reason i haven't extended out to 10 or 15 game history is just the api load.  each request take about 1 second... so we have load times for new games under a few seconds.... but you start adding 15 more games.... that only works with shared cached.... soo that may be a path forward there.


Incorporate Points Per Minute and Quarter Analysis, this is going the direction of higher frequency data.  Like beyond last game stats, could have last game quarterly stats.... Im not sure how well granular stats generalize to game outcomes.... there is a screen shot somewhere of the data feature correlations:
![68747470733a2f2f692e696d6775722e636f6d2f784a4d6d765a522e706e67](https://github.com/nealmick/bb/assets/33704976/e29b5707-c35e-4035-9091-69478518c21b)

the highest coreletating stats are what i would call meta stats.  or long term stats, or stats that incorporate other stats... basically higher level data.  Generally team level stats have highest correlation because it represents a whole team, where as player data is more individual and lower correlation.  Also note streaks highly correlate which is momentum.  The thing about really short term data is it has potential to create alot of noise, and not actually generalize well toward future events.... like its possible some of that data could just not matter and make it more difficult to train..... also this is just like add more basically.... which again, limits to everything... plus none of the data source we using now has that type of data, so would need to provision a new data source for both all our historical games and future data api.....


Projection metrics, this is kinda general, uhm i think game scores are the best target because its decides who actually wins, which is what the players and team and all the data seems to be related to, as opposed to trying to predict closing line values or vegas odds directly.  Plus game scores can be converted to implied plus minus point spread, which can be converted to money line, and also the scores can just be added to together for totals.  which by the way we have absolutely no metrics or evaluation for vegas totals, and thats kinda a completely different thing because its just volume.  although you could easily just add the scores together and call it a total.... thats why we map scores not some other analog soo its all convertible  and the point of the game....  this is actually something i want to work on, i want to build out metrics for totals evaluation.  ive heard totals are easier to predict but never really understood why.  Maybe because higher number more specific better loss calculation, but dont know.


Also evaluation metrics, we have alot of those it can be kinda confusing, we do evaluate models on test hold out data sets the models have never seen before.... that evaluation is displayed on the training page with graphs and variance and all sorts of metrics....  the profile page and the other metrics reflect the users stats on future games... like the dashboard tracks accuracy metrics as you predict games.  This leads to having alot of games and sometime the dashboard docent represent a clean view.... soo i built the bets page.  you can mark games as bets at the bottom of a game, then that game will show in the bets dashboard.  the bets dashboard is supposed to track specific games you actually wanted too, instead of just user account wide stats for every game.... 

iterative testing: one of the features that can help with this is the export data function, if you predict a bunch of games on the dashboard and you want to export a general view of the games, you can click export and it will give you a nice file with general info about all the games on your user account.   The export data function does not include all the data from every game... if you want that its stored by the user name in the csv folder... but the export data function gives you general info so you can pull into excel or whatever and make your own analysis of results really easy....


these are just my initial thoughts ill respond more tomorrow but good ideas
